hal/src/thumbv7em/clock/v2/xosc32k.rs

```diff
@@ -151,109 +153,104 @@ impl Sealed for CrystalMode {}
 // Xosc32k
 //==============================================================================
 
-pub struct Xosc32k<M>
+pub struct Xosc32k<M, X, Y>
 where
     M: Mode,
+    X: Output32k,
+    Y: Output1k,
 {
     token: Xosc32kToken,
-    mode: M,
     xin32: XIn32,
+    mode: M,
+    run_standby: bool,
+    on_demand_mode: bool,
+    start_up_masking: StartUp32k,
+    output32k: PhantomData<X>,
+    output1k: PhantomData<Y>,
 }
 
-impl<M: Mode> Xosc32k<M> {
-    /// TODO
+impl<M, X, Y> Xosc32k<M, X, Y>
+where
+    M: Mode,
+    X: Output32k,
+    Y: Output1k,
+{
+    /// Set for how long the clock output should be masked during startup
     #[inline]
-    pub fn set_start_up(mut self, start_up: StartUp) -> Self {
-        self.token.set_start_up(start_up);
+    pub fn set_start_up(mut self, start_up: StartUp32k) -> Self {
+        self.start_up_masking = start_up;
         self
     }
 
-    /// TODO
+    /// Controls how Xosc32k behaves when a peripheral clock request is detected
     #[inline]
     pub fn set_on_demand(mut self, on_demand: bool) -> Self {
-        self.token.set_on_demand(on_demand);
+        self.on_demand_mode = on_demand;
         self
     }
 
-    /// TODO
+    /// Controls how Xosc32k should behave during standby
     #[inline]
     pub fn set_run_standby(mut self, run_standby: bool) -> Self {
-        self.token.set_run_standby(run_standby);
+        self.run_standby = run_standby;
         self
     }
-
-    /// TODO
-    #[inline]
-    pub fn set_1k_output(mut self, enabled: bool) -> Self {
```

Comment:
Ah, maybe you can't enable both 1k and 32k output simultaneously? That would make sense to prevent at compile-time.